apps/state/lib/state/shape.ex

```diff
@@ -222,16 +222,18 @@ defmodule State.Shape do
     |> Enum.sort_by(& &1.priority, &>=/2)
   end
 
-  defp find_override(id, overrides, suffix_overrides) do
-    if priority = Map.get(overrides, id) do
-      priority
-    else
-      Enum.find_value(suffix_overrides, fn {suffix, priority} ->
-        if String.ends_with?(id, suffix) do
-          priority
-        end
-      end)
-    end
+  defp find_override(id, prefix_overrides, suffix_overrides) do
+    Enum.find_value(prefix_overrides, fn {shape_id, priority} ->
+      if String.starts_with?(id, shape_id) do
+        priority
+      else
+        Enum.find_value(suffix_overrides, fn {suffix, priority} ->
+          if String.ends_with?(id, suffix) do
+            priority
+          end
+        end)
+      end
+    end)
```

Comment:
This is going to loop over all the suffix overrides for each prefix override. You could do something like:
```elixir
Enum.find_value(
  [{&String.starts_with?/2, prefix_overrides}, 
   {&String.ends_with?/2, suffix_overrides}], 
  fn {check_fn, overrides} ->
    Enum.find_value(overrides, fn {check_value, priority} ->
      if check_fn.(id, check_value), do: priority
    end)
  end)
```
which will check all the prefixes, then all the suffixes.